autoload/db/adapter/snowflake.vim

```diff
@@ -0,0 +1,44 @@
+if exists('g:autoloaded_db_snowflake')
+  finish
+endif
+let g:autoloaded_db_snowflake = 1
+
+function! s:command_for_url(url) abort
+  let url = db#url#parse(a:url)
+  "extra options here turn off nonsense before/after results
+  let cmd = (has_key(url, 'password') ? ['env', 'SNOWSQL_PWD=' . url.password] : []) +
+        \ ['snowsql', '-o', 'friendly=false', '-o', 'timing=false'] + 
+        \ db#url#as_argv(a:url, '-a ', '', '', '-u ', '','-d ')
+  for i in keys(url.params)
+    let cmd += ['--'.i.'='.url.params[i]]
+  endfor
+  return cmd
+endfunction
+
+function! db#adapter#snowflake#interactive(url) abort
+  echomsg "starting snowsql. executing query"
+  return s:command_for_url(a:url)
+endfunction
+
+function! db#adapter#snowflake#input_flag() abort
+  return ' -f '
+endfunction
+
+function! db#adapter#snowflake#complete_opaque(url) abort
+  return db#adapter#snowflake#complete_database(url)
+endfunction
+
+function! db#adapter#snowflake#complete_database(url) abort
+  let cmd = s:command_for_url(a:url)
+  let cmd .= ' --query "show databases"'
+  " if cmd looks to have no cli args, treat it as a string
+  let out = system(cmd)
```

Comment:
Needs to be retooled to use `db#systemlist()`.

Reply:
Retooled this, however have to apply a map after calling `systemlist` since snowflake doesn't provide a way to get database names _only_ in the same way e.g. MariaDB does without having an active warehouse.